src/js/main.js

```diff
@@ -0,0 +1,80 @@
+const radTabsElementClassName = '.js-rad-tabs-element';
+const radDropdownElementClassName = '.js-rad-dropdown-element';
+const radContentElementClassName = '.js-rad-content-element';
+
+export const DiscourseRADParser = () => {
+  const radTabsElements = document.querySelectorAll(`${radTabsElementClassName}`);
+
+  radTabsElements.forEach((radTabsElement) => {
+    const radDropdownElements = radTabsElement.querySelectorAll(`${radDropdownElementClassName}`);
+
+    setDefaults(radDropdownElements);
+    setupDropdowns(radDropdownElements);
+  });
+
+  toggleElements();
+}
+
+const setDefaults = radDropdownElements => {
+  radDropdownElements.forEach(radDropdownElement => {
+    const name = radDropdownElement.name;
+    const radDropdownElementOptions = radDropdownElement.querySelectorAll('option');
+
+    let defaultWasSet = false;
+    radDropdownElementOptions.forEach(radDropdownElementOption => {
+      if (localStorage.getItem(name) === radDropdownElementOption.value) {
+        defaultWasSet = true;
+        radDropdownElementOption.setAttribute('selected', 'selected');
+      }
+    });
+
+    if (!defaultWasSet && localStorage.getItem(name) === null) {
+      localStorage.setItem(name, radDropdownElement.firstElementChild.value);
```

Comment:
I am still worried we are setting options here which the user has not selected. For example, if the first element is set as the user option then wouldn't that be surprising when you switch to another page and the selected the option would be the forth element but the user did not make a selection.

Reply:
But we want the settings to persist, don't we? If the user selects `snap 2.7` on `/about`. Wouldn't we want to persist that on all other pages they go to?

The scenario you are describing is something I did not think of. Where `snap 2.7` can be 1st option on one page and 5th option on another.

I would not mind changing this on a future release, if we find it faulty. We can have a chat about it.